libglusterfs/src/dict.c

```diff
@@ -61,6 +61,20 @@ get_new_data()
     return data;
 }
 
+data_t *
+get_new_data_from_pool(glusterfs_ctx_t *ctx)
+{
+    data_t *data = mem_get(ctx->dict_data_pool);
+
+    if (!data)
+        return NULL;
+
+    GF_ATOMIC_INIT(data->refcount, 0);
+    data->is_static = _gf_false;
+
+    return data;
+}
+
```

Comment:
This is exactly the same as `get_new_data()` but with an argument. To avoid code duplication, we should modify `get_new_data()` to simply call this function with the appropriate argument.